src/pages/MyProfile/MyProfile.js

```diff
@@ -1,26 +1,44 @@
 import React, { useState } from 'react'
 import { useQuery } from '@apollo/react-hooks'
 import useForm from 'react-hook-form'
-import HposSettingsQuery from 'graphql/HposSettingsQuery.gql'
+import { useQuery, useMutation } from '@apollo/react-hooks'
 import PrimaryLayout from 'components/layout/PrimaryLayout'
+import HposSettingsQuery from 'graphql/HposSettingsQuery.gql'
+import HposUpdateSettingsMutation from 'graphql/HposUpdateSettingsMutation.gql'
 import Button from 'components/Button'
 import Input from 'components/Input'
 import HashAvatar from 'components/HashAvatar'
 import TosModal from 'components/TosModal'
 import './MyProfile.module.css'
 
 // eslint-disable-next-line no-useless-escape
-const EMAIL_REGEXP = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+// const EMAIL_REGEXP = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+
+// Data - Mutation hook
+function useUpdateDeviceName () {
+  const [hposUpdateSettings] = useMutation(HposUpdateSettingsMutation)
+  return (hostName) => hposUpdateSettings({
+    variables: { hostName }
+  })
+}
 
 const MyProfile = ({ history: { push } }) => {
   const { data: { hposSettings: settings = [] } = {} } = useQuery(HposSettingsQuery)
+  const updateDeviceName = useUpdateDeviceName()
+
   const [isTosOpen, setTosOpen] = useState(false)
   const { register, handleSubmit, errors, watch } = useForm()
   const onSubmit = data => {
+    // console.log('SUBMITTED DATA : ', data)
+    if (data.name) {
+      // call HPOS Settings Mutation to update HPOS Host Name
+      updateDeviceName(data.name)
+    }
+    // TODO : Determine how we would like to handle avatar link data persistance.
```

Comment:
Let's capture this in pivotal and remove the comment

Reply:
:heavy_check_mark: